magma_access_gateway_installer/agw_network_configurator.py

```diff
@@ -0,0 +1,237 @@
+
+        updated_resolved_conf_file = open("/etc/systemd/resolved.conf", "w")
+        updated_resolved_conf_file.writelines(updated_resolved_conf_file_content)
+        updated_resolved_conf_file.close()
+        logger.info("Restarting systemd-resolved service...")
+        check_call(["service", "systemd-resolved", "restart"])
+
+        self.REBOOT_NEEDED = True
+
+    @property
+    def _network_interfaces_config_files_exist(self) -> bool:
+        """Checks whether configuration files for eth0 and eth1 exist in {self.INTERFACES_DIR}."""
+        if os.path.exists(f"{self.INTERFACES_DIR}/eth0") and os.path.exists(
+            f"{self.INTERFACES_DIR}/eth1"
+        ):
+            return True
+        else:
+            return False
+
+    def _create_interfaces_config_files(self):
+        """Configures SGi and S1 AGW interfaces."""
+        logger.info("Configuring SGi and S1 interfaces...")
+        self._prepare_interfaces_configuration_dir_if_doesnt_exist()
+        self._prepare_eth0_configuration()
+        self._prepare_eth1_configuration()
+        self.REBOOT_NEEDED = True
+
+    def _prepare_interfaces_configuration_dir_if_doesnt_exist(self):
+        """Creates a directory to store network interfaces configuration files."""
+        if not os.path.exists(self.INTERFACES_DIR):
+            logger.info(f"Preparing interfaces configuration directory: {self.INTERFACES_DIR}...")
+            os.makedirs(self.INTERFACES_DIR)
+        with open("/etc/network/interfaces", "w") as interfaces_file:
+            interfaces_file.write(f"source-directory {self.INTERFACES_DIR}")
+
+    def _prepare_eth0_configuration(self):
+        """Creates eth0 configuration file under {self.INTERFACES_DIR}/eth0."""
+        logger.info("Preparing configuration for SGi interface (eth0)...")
+        if self.eth0_address:
+            logger.info(
+                f"Using statically configured IP {self.eth0_address} "
+                f"and Gateway {self.eth0_gw_ip_address}..."
+            )
+        else:
+            logger.info("Using DHCP allocated addresses...")
+        with open(f"{self.INTERFACES_DIR}/eth0", "w") as eth0_config_file:
+            eth0_config_file.writelines(line + "\n" for line in self._eth0_config)
+
+        self.REBOOT_NEEDED = True
+
+    def _prepare_eth1_configuration(self):
+        """Creates eth1 configuration file under {self.INTERFACES_DIR}/eth1."""
+        logger.info("Preparing configuration for S1 interface (eth1)...")
+        with open(f"{self.INTERFACES_DIR}/eth1", "w") as eth1_config_file:
+            eth1_config_file.writelines(line + "\n" for line in self._eth1_config)
+
+        self.REBOOT_NEEDED = True
+
+    @property
+    def _networking_service_enabled(self) -> bool:
+        for service in check_output(["systemctl", "list-unit-files"]).decode().splitlines():
+            if all(status in service for status in ["networking.service", "enabled"]):
+                return True
+        return False
```

Comment:
A bit more readable?
```suggestion 
         return any(
            all(status in service for status in ["networking.service", "enabled"])
            for service in check_output(["systemctl", "list-unit-files"])
            .decode()
            .splitlines()
        )
```

Reply:
I like your solution, but I wouldn't say it's more readable. 
Unless convinced otherwise, I'll leave it as is ;)

Reply:
These are my reasons:
1. It's more "pythonic". Idk if that's a heavy enough reason tho
2. Shows at first glance you're looking for a condition to be met at least once. The other way you don't know the overall behavior until you check the code

Idk if I convinced you 😆

Reply:
Yeah, you did ;)